### PR TITLE
Feature/family crud

### PR DESCRIPTION
**Pull request by golabm1993**

### CODE REVIEW

File: src/main/java/pl/golabm/controller/FamilyController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    private final FamilyService familyService;
+
+    @Autowired
+    FamilyController(final FamilyService familyService) {
+        this.familyService = familyService;
+    }
+
+    @PostMapping
+    public Family save(@RequestBody FamilyDTO familyDTO) {
+        Family family = familyDTO.toEntity();
```
Comment (klopucki) [inline src/main/java/pl/golabm/controller/FamilyController.java]: make family final

File: src/main/java/pl/golabm/controller/FamilyController.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @GetMapping
+    public Iterable<Family> getAll() {
+        return familyService.getAll();
+    }
+
+    @PutMapping(value = "/{id}")
+    public Family update(@PathVariable Long id, @RequestBody FamilyDTO familyDTO) {
+        Family family = familyDTO.toEntity();
```
Comment (klopucki) [inline src/main/java/pl/golabm/controller/FamilyController.java]: make family as final

File: src/main/java/pl/golabm/dto/ChildDTO.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        child.setBaptismDate(baptismDate);
+        child.setFirstCommunionDate(firstCommunionDate);
+        child.setConfirmationDate(confirmationDate);
+        child.setSundayHolyMass(sundayHolyMass);
+        child.setConfessionAndHolyCommunion(confessionAndHolyCommunion);
+        child.setParticipationInCatechesis(participationInCatechesis);
+        return child;
+    }
+
+    public static ChildDTO fromEntity(Child child) {
```
Comment (klopucki) [inline src/main/java/pl/golabm/dto/ChildDTO.java]: make child method property as final

File: src/main/java/pl/golabm/dto/ChildDTO.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        ChildDTO childDTO = new ChildDTO();
```
Comment (klopucki) [inline src/main/java/pl/golabm/dto/ChildDTO.java]: make it final

File: src/main/java/pl/golabm/dto/ChildDTO.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    public String getParticipationInCatechesis() {
+        return participationInCatechesis;
+    }
+
+    public void setParticipationInCatechesis(String participationInCatechesis) {
+        this.participationInCatechesis = participationInCatechesis;
+    }
+
+    public Child toEntity() {
+        Child child = new Child();
```
Comment (klopucki) [inline src/main/java/pl/golabm/dto/ChildDTO.java]: make it final

File: src/main/java/pl/golabm/dto/FamilyDTO.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    public Date getLastPastoralVisit() {
+        return lastPastoralVisit;
+    }
+
+    public void setLastPastoralVisit(Date lastPastoralVisit) {
+        this.lastPastoralVisit = lastPastoralVisit;
+    }
+
+    public Family toEntity() {
+        Family family = new Family();
```
Comment (klopucki) [inline src/main/java/pl/golabm/dto/FamilyDTO.java]: make it final

File: src/main/java/pl/golabm/dto/FamilyDTO.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        List<FamilyMember> familyMember = familyMembers.stream().map(FamilyMemberDTO::toEntity).collect(Collectors.toList());
```
Comment (klopucki) [inline src/main/java/pl/golabm/dto/FamilyDTO.java:86]: I LIKE IT :)

File: src/main/java/pl/golabm/dto/FamilyDTO.java [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+        family.setAddress(address);
+        family.setPhoneNumber(phoneNumber);
+        family.setFamilyMembers(familyMember);
+        family.setChildList(children);
+        family.setLastPastoralVisit(lastPastoralVisit);
+        return family;
+    }
+
+    public FamilyDTO fromEntity(Family family) {
+        FamilyDTO familyDTO = new FamilyDTO();
```
Comment (klopucki) [inline src/main/java/pl/golabm/dto/FamilyDTO.java]: make it final

File: src/main/java/pl/golabm/service/FamilyService.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    public void delete(Long id) {
+        familyRepository.deleteById(id);
+    }
+
+    public Iterable<Family> getAll() {
+        return familyRepository.findAll();
+    }
+
+    public Family update(Long id, Family family) {
+        Optional<Family> oldFammily = familyRepository.findById(id);
```
Comment (klopucki) [inline src/main/java/pl/golabm/service/FamilyService.java]: Set id to family property and use familyRepository.save(family). This should automatically update all props.

Comment (golabm1993) [reply src/main/java/pl/golabm/service/FamilyService.java]: Could you explain that more? How to set id to family property?

Comment (klopucki) [reply src/main/java/pl/golabm/service/FamilyService.java]: family.setid(id);
return familyRepository.save(family);

:)

**Review by klopucki — APPROVED** [review]

**Pull request merged**